### PR TITLE
Nomad: updating max token length to 256

### PR DESCRIPTION
**Pull request by burdandrei**
as it's updated in Nomad 0.8
https://github.com/hashicorp/nomad/pull/3888

### CODE REVIEW

**Comment by jefferai** [comment]
I'm not sure we want to do this yet since most people won't be running Nomad 0.8 for a while.

**Comment by burdandrei** [comment]
@jefferai you're right, just leave it here for the future =)

**Comment by jippi** [comment]
they will pretty soon though, like in the next few weeks :)

**Comment by burdandrei** [comment]
if you want I can add the configuration option with the target nomad version

**Comment by jippi** [comment]
does it matter? isn't it just truncating a key - so if you pass in 64 (nomad <.0.8) it just won't truncate?

**Comment by jefferai** [comment]
Yeah, rather than target Nomad version you'd want to just specify a max length, which can default to 64. At some point we can switch that to default to 256.

**Comment by burdandrei** [comment]
I think normal operator doesn't have to know there are those limits, like MySQL plugin(legacy, rds, Maria)

File: builtin/logical/nomad/path_creds_create.go [hunk 1]
```diff
@@ -58,8 +58,9 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 
 	// Handling nomad maximum token length
 	// https://github.com/hashicorp/nomad/blob/d9276e22b3b74674996fb548cdb6bc4c70d5b0e4/nomad/structs/structs.go#L115
-	if len(tokenName) > 64 {
-		tokenName = tokenName[0:63]
+	// size increased in https://github.com/hashicorp/nomad/pull/3888
```
Comment (kalafut) [inline builtin/logical/nomad/path_creds_create.go]: Probably not needed as `git blame` will find it. Or if the increase is important to be aware of, then add some explanatory text.

File: builtin/logical/nomad/path_creds_create.go [hunk 2]
```diff
@@ -58,8 +58,9 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
...
+	if len(tokenName) > 256 {
+		tokenName = tokenName[0:255]
```
Comment (kalafut) [inline builtin/logical/nomad/path_creds_create.go]: This is fine; `[:255]` is more common.

**Comment by chrishoffman** [comment]
Fixed in #5117

**Pull request closed without merging**